diff -Naupr p7zip_4.51.orig/makefile.linux_amd64_asm p7zip_4.51/makefile.linux_amd64_asm
--- p7zip_4.51.orig/makefile.linux_amd64_asm	2007-06-28 09:32:30.000000000 +0200
+++ p7zip_4.51/makefile.linux_amd64_asm	2007-08-05 17:52:20.000000000 +0200
@@ -2,7 +2,7 @@
 OPTFLAGS=-O
 
 # use "-m32" to have a 32bits executable
-ALLFLAGS=-m64 ${OPTFLAGS} -s \
+ALLFLAGS=-m64 ${OPTFLAGS} \
     	-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE \
 	-DNDEBUG -D_REENTRANT -DENV_UNIX \
 	$(LOCAL_FLAGS)
diff -Naupr p7zip_4.51.orig/makefile.linux_x86_ppc_alpha_gcc_4.X p7zip_4.51/makefile.linux_x86_ppc_alpha_gcc_4.X
--- p7zip_4.51.orig/makefile.linux_x86_ppc_alpha_gcc_4.X	2007-06-28 09:33:43.000000000 +0200
+++ p7zip_4.51/makefile.linux_x86_ppc_alpha_gcc_4.X	2007-08-05 17:52:45.000000000 +0200
@@ -5,7 +5,7 @@
 
 OPTFLAGS=-O
 
-ALLFLAGS=${OPTFLAGS} -s \
+ALLFLAGS=${OPTFLAGS} \
     	-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE \
 	-DNDEBUG -D_REENTRANT -DENV_UNIX \
 	$(LOCAL_FLAGS)
