diff -Naupr p7zip_4.61.orig/CPP/7zip/Bundles/Format7zFree/makefile p7zip_4.61/CPP/7zip/Bundles/Format7zFree/makefile
--- p7zip_4.61.orig/CPP/7zip/Bundles/Format7zFree/makefile	2008-11-28 21:17:16.000000000 +0100
+++ p7zip_4.61/CPP/7zip/Bundles/Format7zFree/makefile	2008-12-23 16:27:04.927434325 +0100
@@ -83,9 +83,6 @@ HmacSha1.o  \
 Pbkdf2HmacSha1.o  \
 RandGen.o  \
 Sha1.o  \
-RarAES.o  \
-Rar20Cipher.o  \
-Rar20Crypto.o  \
 WzAES.o  \
 ZipCipher.o  \
 ZipCrypto.o  \
@@ -124,12 +121,6 @@ StreamUtils.o  \
 7zRegister.o  \
 7zSpecStream.o  \
 7zUpdate.o  \
-RarHandler.o  \
-RarHeader.o  \
-RarIn.o  \
-RarItem.o  \
-RarRegister.o  \
-RarVolumeInStream.o  \
 bz2Register.o  \
 BZip2Handler.o  \
 BZip2HandlerOut.o  \
diff -Naupr p7zip_4.61.orig/CPP/7zip/Bundles/Format7zFree/makefile.list p7zip_4.61/CPP/7zip/Bundles/Format7zFree/makefile.list
--- p7zip_4.61.orig/CPP/7zip/Bundles/Format7zFree/makefile.list	2008-08-14 16:45:27.000000000 +0200
+++ p7zip_4.61/CPP/7zip/Bundles/Format7zFree/makefile.list	2008-12-23 16:28:13.942309528 +0100
@@ -51,11 +51,6 @@ SRCS=\
  ../../Compress/BZip2/BZip2Decoder.cpp \
  ../../Compress/BZip2/BZip2Encoder.cpp \
  ../../Compress/BZip2/BZip2Register.cpp \
- ../../Compress/Rar/Rar1Decoder.cpp \
- ../../Compress/Rar/Rar2Decoder.cpp \
- ../../Compress/Rar/Rar3Decoder.cpp \
- ../../Compress/Rar/Rar3Vm.cpp \
- ../../Compress/Rar/RarCodecsRegister.cpp \
  ../../Compress/Implode/ImplodeDecoder.cpp \
  ../../Compress/Implode/ImplodeHuffmanDecoder.cpp \
  ../../Compress/Lzx/Lzx86Converter.cpp \
@@ -72,9 +67,6 @@ SRCS=\
  ../../Crypto/Hash/Pbkdf2HmacSha1.cpp \
  ../../Crypto/Hash/RandGen.cpp \
  ../../Crypto/Hash/Sha1.cpp \
- ../../Crypto/RarAES/RarAES.cpp \
- ../../Crypto/Rar20/Rar20Cipher.cpp \
- ../../Crypto/Rar20/Rar20Crypto.cpp \
  ../../Crypto/WzAES/WzAES.cpp \
  ../../Crypto/Zip/ZipCipher.cpp \
  ../../Crypto/Zip/ZipCrypto.cpp \
@@ -113,12 +105,6 @@ SRCS=\
  ../../Archive/7z/7zRegister.cpp \
  ../../Archive/7z/7zSpecStream.cpp \
  ../../Archive/7z/7zUpdate.cpp \
- ../../Archive/Rar/RarHandler.cpp \
- ../../Archive/Rar/RarHeader.cpp \
- ../../Archive/Rar/RarIn.cpp \
- ../../Archive/Rar/RarItem.cpp \
- ../../Archive/Rar/RarRegister.cpp \
- ../../Archive/Rar/RarVolumeInStream.cpp \
  ../../Archive/BZip2/bz2Register.cpp \
  ../../Archive/BZip2/BZip2Handler.cpp \
  ../../Archive/BZip2/BZip2HandlerOut.cpp \
@@ -347,16 +333,6 @@ BZip2Encoder.o : ../../Compress/BZip2/BZ
 	$(CXX) $(CC_SHARED) $(CFLAGS) ../../Compress/BZip2/BZip2Encoder.cpp
 BZip2Register.o : ../../Compress/BZip2/BZip2Register.cpp
 	$(CXX) $(CC_SHARED) $(CFLAGS) ../../Compress/BZip2/BZip2Register.cpp
-Rar1Decoder.o : ../../Compress/Rar/Rar1Decoder.cpp
-	$(CXX) $(CC_SHARED) $(CFLAGS) ../../Compress/Rar/Rar1Decoder.cpp
-Rar2Decoder.o : ../../Compress/Rar/Rar2Decoder.cpp
-	$(CXX) $(CC_SHARED) $(CFLAGS) ../../Compress/Rar/Rar2Decoder.cpp
-Rar3Decoder.o : ../../Compress/Rar/Rar3Decoder.cpp
-	$(CXX) $(CC_SHARED) $(CFLAGS) ../../Compress/Rar/Rar3Decoder.cpp
-Rar3Vm.o : ../../Compress/Rar/Rar3Vm.cpp
-	$(CXX) $(CC_SHARED) $(CFLAGS) ../../Compress/Rar/Rar3Vm.cpp
-RarCodecsRegister.o : ../../Compress/Rar/RarCodecsRegister.cpp
-	$(CXX) $(CC_SHARED) $(CFLAGS) ../../Compress/Rar/RarCodecsRegister.cpp
 ImplodeDecoder.o : ../../Compress/Implode/ImplodeDecoder.cpp
 	$(CXX) $(CC_SHARED) $(CFLAGS) ../../Compress/Implode/ImplodeDecoder.cpp
 ImplodeHuffmanDecoder.o : ../../Compress/Implode/ImplodeHuffmanDecoder.cpp
@@ -391,12 +367,6 @@ RandGen.o : ../../Crypto/Hash/RandGen.cp
 	$(CXX) $(CC_SHARED) $(CFLAGS) ../../Crypto/Hash/RandGen.cpp
 Sha1.o : ../../Crypto/Hash/Sha1.cpp
 	$(CXX) $(CC_SHARED) $(CFLAGS) ../../Crypto/Hash/Sha1.cpp
-RarAES.o : ../../Crypto/RarAES/RarAES.cpp
-	$(CXX) $(CC_SHARED) $(CFLAGS) ../../Crypto/RarAES/RarAES.cpp
-Rar20Cipher.o : ../../Crypto/Rar20/Rar20Cipher.cpp
-	$(CXX) $(CC_SHARED) $(CFLAGS) ../../Crypto/Rar20/Rar20Cipher.cpp
-Rar20Crypto.o : ../../Crypto/Rar20/Rar20Crypto.cpp
-	$(CXX) $(CC_SHARED) $(CFLAGS) ../../Crypto/Rar20/Rar20Crypto.cpp
 WzAES.o : ../../Crypto/WzAES/WzAES.cpp
 	$(CXX) $(CC_SHARED) $(CFLAGS) ../../Crypto/WzAES/WzAES.cpp
 ZipCipher.o : ../../Crypto/Zip/ZipCipher.cpp
@@ -473,18 +443,6 @@ StreamUtils.o : ../../Common/StreamUtils
 	$(CXX) $(CC_SHARED) $(CFLAGS) ../../Archive/7z/7zSpecStream.cpp
 7zUpdate.o : ../../Archive/7z/7zUpdate.cpp
 	$(CXX) $(CC_SHARED) $(CFLAGS) ../../Archive/7z/7zUpdate.cpp
-RarHandler.o : ../../Archive/Rar/RarHandler.cpp
-	$(CXX) $(CC_SHARED) $(CFLAGS) ../../Archive/Rar/RarHandler.cpp
-RarHeader.o : ../../Archive/Rar/RarHeader.cpp
-	$(CXX) $(CC_SHARED) $(CFLAGS) ../../Archive/Rar/RarHeader.cpp
-RarIn.o : ../../Archive/Rar/RarIn.cpp
-	$(CXX) $(CC_SHARED) $(CFLAGS) ../../Archive/Rar/RarIn.cpp
-RarItem.o : ../../Archive/Rar/RarItem.cpp
-	$(CXX) $(CC_SHARED) $(CFLAGS) ../../Archive/Rar/RarItem.cpp
-RarRegister.o : ../../Archive/Rar/RarRegister.cpp
-	$(CXX) $(CC_SHARED) $(CFLAGS) ../../Archive/Rar/RarRegister.cpp
-RarVolumeInStream.o : ../../Archive/Rar/RarVolumeInStream.cpp
-	$(CXX) $(CC_SHARED) $(CFLAGS) ../../Archive/Rar/RarVolumeInStream.cpp
 bz2Register.o : ../../Archive/BZip2/bz2Register.cpp
 	$(CXX) $(CC_SHARED) $(CFLAGS) ../../Archive/BZip2/bz2Register.cpp
 BZip2Handler.o : ../../Archive/BZip2/BZip2Handler.cpp
diff -Naupr p7zip_4.61.orig/makefile p7zip_4.61/makefile
--- p7zip_4.61.orig/makefile	2008-11-30 16:49:38.000000000 +0100
+++ p7zip_4.61/makefile	2008-12-23 16:28:49.215386938 +0100
@@ -42,7 +42,6 @@ depend:
 	cd CPP/7zip/UI/Console        ; $(MAKE) depend
 	cd CPP/7zip/UI/GUI            ; $(MAKE) depend
 	cd CPP/7zip/Bundles/Format7zFree ; $(MAKE) depend
-	cd CPP/7zip/Compress/Rar      ; $(MAKE) depend
 
 sfx: common
 	$(MKDIR) bin
@@ -51,7 +50,6 @@ sfx: common
 common7z:common
 	$(MKDIR) bin/Codecs
 	cd CPP/7zip/Bundles/Format7zFree ; $(MAKE) all
-	cd CPP/7zip/Compress/Rar         ; $(MAKE) all
 
 7z: common7z
 	cd CPP/7zip/UI/Console           ; $(MAKE) all
@@ -71,7 +69,6 @@ clean:
 	cd CPP/7zip/UI/FileManager       ; $(MAKE) clean
 	cd CPP/7zip/UI/GUI               ; $(MAKE) clean
 	cd CPP/7zip/Bundles/Format7zFree ; $(MAKE) clean
-	cd CPP/7zip/Compress/Rar         ; $(MAKE) clean
 	cd CPP/7zip/Compress/LZMA_Alone  ; $(MAKE) clean
 	cd CPP/7zip/Compress/PPMD_Alone  ; $(MAKE) clean
 	cd CPP/7zip/Bundles/AloneGCOV    ; $(MAKE) clean
